### ISSUE TITLE
Web server non-JSON response?

### ISSUE THREAD

**Issue by balthisar** [validator/validator#627]
According to the [documentation](https://github.com/validator/validator/wiki/Service-%C2%BB-HTTP-interface#output-modes), I should be able to specify different output modes, but I don't see how to make it work for anything other than JSON.

With both my local server as well as the w3c-hosted server, I'm getting:

~~~
  NSErrorFailingURLKey = http://localhost:8888/?out=xml
NSLocalizedDescription = Request failed: unacceptable content-type: application/xml
           status code = 200
             localized = no error
~~~

The only one that seems to work for me is `out=json` (which suits my needs), but if I'd wanted to switch to XML, I'd be scratching my head even harder right now. I'm pretty sure it's nothing in the POST headers:

~~~
    Accept = "text/xml";
    "Accept-Encoding" = gzip;
    "Content-Length" = 175;
    "Content-Type" = "text/html; charset=utf-8";
~~~

I've also tried eliminating the `accept` header, as well as changing the value to `application/xml` to no avail.

Am I doing something wrong, or are the other output modes disabled on the server right now?

Thanks.

**Comment by sideshowbarker** [comment]
There’s nothing disabled on the server, so I’m not sure what might be causing the error cited in the issue description. *“unacceptable content-type”* isn’t a message generated by the checker code.

Anyway, the following is an example of using `curl` to send a POST with `out=xml` requested:

```
curl -H 'Content-Type: text/html; charset=utf-8' \
    --data-binary '<!doctype html><html lang=en><title>test</title><foo></foo>' \
    https://validator.w3.org/nu/?out=xml
```

And the following is the XML output that the above request returns:

```xml
<?xml version='1.0' encoding='utf-8'?>
<messages xmlns="http://n.validator.nu/messages/">
<error last-line="1" last-column="53" first-column="49">
<message>Element <code xmlns="http://www.w3.org/1999/xhtml">foo</code> not allowed as child of element <a xmlns="http://www.w3.org/1999/xhtml" href="http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-body-element"><code>body</code></a> in this context. (Suppressing further errors from this subtree.)</message>
<extract>st&lt;/title&gt;<m>&lt;foo&gt;</m>&lt;/foo&gt;</extract><elaboration><dl xmlns="http://www.w3.org/1999/xhtml"><dt>Content model for element <a href="http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-body-element"><code>body</code></a>:</dt><dd><a href="http://www.w3.org/html/wg/drafts/html/master/single-page.html#flow-content-2">Flow content</a>.
      </dd></dl></elaboration>
</error>

</messages>
```

**Comment by balthisar** [comment]
Thanks, @sideshowbarker. I'll dig into my issue deeper, and close this out.
